### PR TITLE
erasure: Handle failed disks so that we initialize properly if they a…

### LINKED ISSUE #1579: XL: minio server allows to run with wrong configuration (May 11th)
Steps to reproduce
1. run minio server

``` shell
minio server --address=:60000 /mnt/sd{b..c}1/bala/mydisk
```
1. stop the server and run again with

``` shell
minio server --address=:60000 /mnt/sd{b..e}1/bala/mydisk
```

It works!
1. run minio server again with

``` shell
minio server --address=:60000 /mnt/sd{b..c}1/bala/mydisk
```

This time below errors are thrown

```
ERRO[0000] MakeVol failed with volume already exists     volume=.minio
ERRO[0000] MakeVol failed with volume already exists     volume=.minio
ERRO[0000] Verification failed after reconstruction, data likely corrupted.  path=format.json volume=.minio
ERRO[0000] loadFormatXL failed with Verification failed after reconstruction, data likely corrupted.
FATA[0000] Initializing object layer failed.             Error={Cause:Command-line arguments [/mnt/sdb1/bala/mydisk /mnt/sdc1/bala/mydisk] is not valid. Type:*errors.errorString}

```

Below is minio version

```
$ minio version
Version: 2016-05-10T13:34:20Z
Release-Tag: DEVELOPMENT.2016-05-10T13-34-20Z
Commit-ID: 409e09c1e55ab448419a3d62cd8a5704312b2172
```

### LINKED ISSUE #1592: XL: minio server does not allow to run with failed disk upto the read/write quorum (May 11th)
Steps to reproduce
1. run minio server with 8 disks

``` shell
minio server /mnt/sd{b..i}1/bala/mydisk
```
1. stop the server
2. remove backend disk `/mnt/sdi1/bala/mydisk
3. run minio server again

``` shell
minio server /mnt/sd{b..i}1/bala/mydisk
```

It stops running with following error

```
FATA[0000] Initializing object layer failed.             Error={Cause:stat /mnt/sdi1/bala/mydisk: no such file or directory Type:*os.PathError}
```

### PR DESCRIPTION
**Pull request by harshavardhana**
…re missing.

Fixes #1592
Fixes #1579

### CODE REVIEW

File: rpc-server.go [hunk 1]
```diff
@@ -128,7 +128,10 @@ func (s *storageServer) RenameFileHandler(arg *RenameFileArgs, reply *GenericRep
 // Initialize new storage rpc.
 func newRPCServer(exportPath string) (*storageServer, error) {
 	// Initialize posix storage API.
-	storage, err := newPosix(exportPath)
+	// Ignore disk errors and successfully initialize posix, errors
+	// will be handled by top level caller.
+	ignoreDiskErr := true
+	storage, err := newPosix(exportPath, ignoreDiskErr)
```
Comment (balamurugana) [inline rpc-server.go]: I think it would be better to handle error and have ignoreDiskErr here than at newPosix()

File: xl-erasure-v1-errors.go [hunk 2]
```diff
@@ -19,10 +19,13 @@ package main
...
 // errMaxDisks - returned for reached maximum of disks.
-var errMaxDisks = errors.New("Total number of disks specified is higher than supported maximum of '16'")
+var errMaxDisks = errors.New("Number of disks on command line is higher than supported maximum of '16'")
+
+// errMaxDisks - returned for minimum number of disks.
+var errMinDisks = errors.New("Number of disks on command line is smaller than supported minimum of '8'")
 
 // errNumDisks - returned for odd numebr of disks.
-var errNumDisks = errors.New("Invalid number of disks provided, should be always multiples of '2'")
+var errNumDisks = errors.New("Invalid number of disks command line, should be always multiples of '2'")
```
Comment (balamurugana) [inline xl-erasure-v1-errors.go]: Its better not to mention "command line" here.  We could say something like "Number of disks should be multiples of 2"

File: xl-erasure-v1-errors.go [hunk 3]
```diff
@@ -19,10 +19,13 @@ package main
 import "errors"
 
 // errMaxDisks - returned for reached maximum of disks.
-var errMaxDisks = errors.New("Total number of disks specified is higher than supported maximum of '16'")
+var errMaxDisks = errors.New("Number of disks on command line is higher than supported maximum of '16'")
```
Comment (balamurugana) [inline xl-erasure-v1-errors.go]: "Number of disks are higher than maximum supported count '16'"

File: xl-erasure-v1-errors.go [hunk 4]
```diff
@@ -19,10 +19,13 @@ package main
...
+
+// errMaxDisks - returned for minimum number of disks.
```
Comment (balamurugana) [inline xl-erasure-v1-errors.go]: comment needs to be fixed.

File: xl-erasure-v1-errors.go [hunk 5]
```diff
@@ -19,10 +19,13 @@ package main
...
+var errMinDisks = errors.New("Number of disks on command line is smaller than supported minimum of '8'")
```
Comment (balamurugana) [inline xl-erasure-v1-errors.go]: "Number of disks are smaller than minimum supported count '8'"

**Comment by harshavardhana** [comment]
> I think it would be better to handle error and have ignoreDiskErr here than at newPosix()

The reason to do not do that newPosix() does not initialize in case of errors, we need an initialized version of POSIX storageAPI ignoring the disks errors. So that we can perform operation and validate. 

Another option is always intialize even in case of errors and let the caller decide if they wish to use storageAPI or not.

**Pull request merged**